### ISSUE TITLE
Error: Model Property does not have relationship \"work_orders\"."

### ISSUE THREAD

**Issue by somecallmemike** [keithwhor/nodal#338]
I am attempting to create a relationship between a single 'property' and multiple 'work_orders', however not matter how I configure the system it is not working.  I am getting the error Error: Model Property does not have relationship \"work_orders\"." using the following configuration:

relationships.js:
```
    const WorkOrder = Nodal.require('app/models/work_order.js');
        
    WorkOrder.joinsTo(Property, {multiple: true});
```
properties_controller.js:

```
    Property.query()
      .where(this.params.query)
      .join('work_orders')
      .end((err, models) => {
        this.respond(err || models,  [
          'id', 
          'name',
          'address',
          'city',
          'county',
          'state',
          'zip',
          'country',
          {'work_orders': 
            [
              'id',
              'name',
              'description'
            ]
          }
        ]);

      });
```

**Comment by jacoblee93** [comment]
Hi Mike,

I have a couple of ideas:

Is there a property on work_orders called property_id? Nodal will attempt to use that to join the two models.

Are you requiring property in your relationships.js file? What about requiring your relationships.js file in the file you're performing the query?

Otherwise, you could try putting the join method before the where method in your query?

Let me know if one of those solves your issue!

Best,

Jacob

**Comment by somecallmemike** [comment]
Jacob,

Thanks for the quick response.  Answers inline:

Is there a property on work_orders called property_id? Nodal will attempt to use that to join the two models.  

Yes, I have a column named property_id in the work_orders table, here is the schema:

```
    "WorkOrder": {

      "table": "work_orders",

      "columns": [
        {"name": "id", "type": "serial", "properties": 
        {"nullable":false,"primary_key":true,"auto_increment":true}},
        {"name": "property_id", "type": "int", "properties": {}},
        {"name": "tenant_id", "type": "int", "properties": {}},
        {"name": "assigned_to", "type": "int", "properties": {}},
        {"name": "name", "type": "string", "properties": {}},
        {"name": "description", "type": "string", "properties": {}},
        {"name": "budget", "type": "int", "properties": {}},
        {"name": "completed_at", "type": "datetime", "properties": {}},
        {"name": "status", "type": "int", "properties": {}},
        {"name": "created_at", "type": "datetime", "properties": {}},
        {"name": "updated_at", "type": "datetime", "properties": {}}
      ]

    },
```

Are you requiring property in your relationships.js file? What about requiring your relationships.js file in the file you're performing the query? 

Yes, property is being required.  Here is the contents of the relationships file:

```
const User = Nodal.require('app/models/user.js');
const Company = Nodal.require('app/models/company.js');
const Property = Nodal.require('app/models/property.js');
const Tenant = Nodal.require('app/models/tenant.js');
const WorkOrder = Nodal.require('app/models/work_order.js');

User.joinsTo(Company, {multiple: false});
Property.joinsTo(User, {multiple: false});
Tenant.joinsTo(Property, {multiple: true});
WorkOrder.joinsTo(Property, {multiple: true});
```

I also tried removing the relationship for Tenant to see if it was conflicting in some way, which also did not work.

Otherwise, you could try putting the join method before the where method in your query?

Just tried this and no luck.  

To add, the Tenant relationship works just fine.  The only difference I can tell is that the string I am using to relate tenants does not include an underscore, and I had to change it from tenant to tenants when defining {multiple: true}.  Attempting to change work_order to work_orders, or workorders, or WorkOrders, or workOrders also fail.

Thanks again!

**Comment by jacoblee93** [comment]
Hi Mike,

It actually should be workOrders, I forgot.

I was able to repro by setting up a simple project. If changing the join to workOrders isn't working, you might not be requiring your `relationships.js` file in your controller. Try putting:

```
const Relationships = Nodal.require('app/relationships.js');
```

at the top of your `properties_controller.js` file? Let me know.

Best,

Jacob

**Comment by somecallmemike** [comment]
Jacob,

Much thanks, setting the join string to 'workOrders' is what fixed it, I guess I had not tried that.  Also I did not need to include relationships.js in the properties_controller.js file in order to get it working.

Best Regards!

Michael

**Comment by jacoblee93** [comment]
Awesome! You should require relationships in your controllers as best practice -- you're probably requiring it in some other file, but if it's not present and you try to do a join, it will fail.

**Comment by jacoblee93** [comment]
Closes #338
